tests/test_openjdk.py

```diff
@@ -152,3 +172,70 @@ def test_jdk_extended(
 
     for check in params.expected_err_strings:
         assert check in testout.stderr
+
+
+@pytest.mark.parametrize(
+    "container_per_test",
+    CONTAINER_IMAGES_CASSANDRA,
+    indirect=["container_per_test"],
+)
+def test_jdk_cassandra(container_per_test):
+    """Starts the Cassandra DB and executes some write and read tests
+    using the cassandra-stress
+    """
+
+    cassandra_versions = container_per_test.connection.check_output(
+        "git ls-remote --tags https://gitbox.apache.org/repos/asf/cassandra.git"
+    )
+
+    lines = re.split("\n", cassandra_versions)
+    version = [0, 0, 0]
+    for line in lines:
+        x = re.search(r"cassandra-(\d).(\d).(\d)$", line)
+        if x:
+            if Version(
+                int(x.group(1)), int(x.group(2)), int(x.group(3))
+            ) > Version(int(version[0]), int(version[1]), int(version[2])):
+                version[0] = x.group(1)
+                version[1] = x.group(2)
+                version[2] = x.group(3)
+
+    cassandra_version = ".".join(version)
```

Comment:
I think something like this should work and is much simpler:
```suggestion
    cassandra_version = Version(0, 0, 0)
    for line in cassandra_versions.splitlines():
        cur_ver = Version.parse(line.replace("cassandra-", ""))
        if cur_ver > cassandra_version:
            cassandra_version = cur_ver
```

Reply:
Done with some changes in your code.
`cur_ver = Version.parse(line.replace("cassandra-", ""))` doesn't work because some special cases. e.g.
`ValueError: Invalid version string: 66ee8b58577a6b61c1195eaa797d14ce38042ea7	refs/tags/0.3.0-final`

Reply:
Ah, yeah, makes sense, I did not actually try my code code :sweat_smile: